File: 242.valid-anagram.java
/*
 * @lc app=leetcode id=242 lang=java
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] charArray = new int[26];
        char[] array1 = s.toCharArray();
        for (char c : array1) {
            charArray[c - 'a']++;
        }

        for (char c : t.toCharArray()) {
            charArray[c - 'a']--;
            if (charArray[c - 'a'] < 0) {
                return false;
            }
        }

        for (int i = 0; i < charArray.length; i++) {
            if (charArray[i] > 0)
                return false;
        }

        return true;
    }
}
// @lc code=end

File: 242.valid-anagram.js
/*
 * @lc app=leetcode id=242 lang=javascript
 *
 * [242] Valid Anagram
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length != t.length) {
    return false;
  }
  let table = new Array(26).fill(0);
  for (let i = 0; i < s.length; i++) {
    table[s.charCodeAt(i) - "a".charCodeAt(0)] += 1;
  }

  for (let i = 0; i < t.length; i++) {
    table[t.charCodeAt(i) - "a".charCodeAt(0)] -= 1;
    if (table[t.charCodeAt(i) - "a".charCodeAt(0)] < 0) {
      return false;
    }
  }
  return true;
};
// @lc code=end
